src/lib/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -1445,6 +1445,148 @@ public function loadRelations(
         return $query->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    public function countRelations(
+        int $contentId,
+        ?int $contentVersionNo = null,
+        ?int $relationType = null
+    ): int {
+        $query = $this->connection->createQueryBuilder();
+        $expr = $query->expr();
+        $query
+            ->select($this->databasePlatform->getCountExpression('l.id'))
+            ->from(self::CONTENT_RELATION_TABLE, 'l')
+            ->innerJoin(
+                'l',
+                'ezcontentobject',
+                'ezcontentobject_to',
+                $expr->and(
+                    'l.to_contentobject_id = ezcontentobject_to.id',
+                    'ezcontentobject_to.status = :status'
+                )
+            )
+            ->where(
```

Comment:
Prefer `andWhere` over `where`. This may be important in cases where query is built with already existing conditions.

```suggestion
            ->andWhere(
```